File: problem003/3.pl
#!/usr/bin/env perl
use strict;
use warnings;

my $n = 600851475143;
my $i = 2;

sub factor {
	$n = shift;
	$i = shift;
	my $maxfactor = 2;
	while ($i < $n/2) {
		if ($n % $i == 0) {
			$n = $n / $i;
			$maxfactor = $n;
		} else {
			$i++;
		}
	}
	return $maxfactor;
}
print(factor($n, $i));

File: problem003/3.lua
#!/usr/bin/env lua
n = 600851475143
i = 2

function factor(n, i)
	maxfactor = 2
	while i < n / 2 do
		if n % i == 0 then
			n = n / i
			maxfactor = n
		else
			i = i + 1
		end
	end
	return maxfactor
end

print(factor(n, i))
